flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/async/AsyncFunction.java

```diff
@@ -89,6 +89,7 @@
 	/**
 	 * {@link AsyncFunction#asyncInvoke} timeout occurred.
 	 * By default, the result future is exceptionally completed with a timeout exception.
+	 * Note that timeout and asyncInvoke might be invoked at the same time.
```

Comment:
I also thought about it, but the question is on how you define invoke:
if I have an asyncInvoke to an external system that takes a long time, then the timeout will be called when asyncInvoke is still in execution. so both are active at the same time, but ofc if you define invocation as the act of trigger the call, then it's not possible.

Reply:
This method is currently supposed to be called in the task's main thread. It may happen that the timer would be triggered during `asyncInvoke()` call.
But (with the current changes in the `master` branch) the trigger itself won't call the `timeout()`, it would just post a mailbox letter to delegate the call to the task's main thread.
Before, this comment made sense, because both methods were called from different (independent) threads.

From API usage point of view, the `asyncInvoke()` is supposed to be "lightweight", it should initiate an actual asynchronous operation (maybe using some user defined execution pool) and give control back. Otherwise, it won't differ from normal "flat map" operator and would block the task's progress.

Reply:
Okay I'm removing it.